benchmark/atoi.cpp

```diff
@@ -117,29 +207,47 @@ std::size_t c_strLength_manualComparison(const char *s) {
     }
 }
 
+#if ZOO_CONFIGURED_TO_USE_AVX()
+
+/// \note Partially generated by Chat GPT 4
 size_t avx2_strlen(const char* str) {
     const __m256i zero = _mm256_setzero_si256(); // Vector of 32 zero bytes
     size_t offset = 0;
+    __m256i data;
+    auto [alignedBase, misalignment] = blockAlignedLoad(str, &data);
 
-    // Loop over the string in blocks of 32 bytes
-    for (;; offset += 32) {
-        // Load 32 bytes of the string into a __m256i vector
-        __m256i data;// = _mm256_load_si256((const __m256i*)(str + offset));
-        memcpy(&data, str + offset, 32);
-        // Compare each byte with '\0'
-        __m256i cmp = _mm256_cmpeq_epi8(data, zero);
-        // Create a mask indicating which bytes are '\0'
-        int mask = _mm256_movemask_epi8(cmp);
+    // AVX does not offer a practical way to generate a mask of all ones in
+    // the least significant positions, thus we cant invoke adjustFor_strlen.
+    // We will do the first iteration as a special case to explicitly take into
+    // account misalignment
```

Comment:
Were any of the comments generated by ChatGPT? I think that could be good to know too.

Reply:
I think I opened a can of worms by crediting ChatGPT 4.
Yes, some comments are from the tool, the others mine, and I forgot which is who's.
What's your opinion about crediting the AI? I think it is sufficient to indicate that a tool was involved and the result is hybrid.
What I wanted was to let the reader know I did not have to do the heavy lifting of selecting the intrinsics.
The ARM/NEON flavor was a different experience, though, I did not manage to get usable code from the tool, I had to decode the GLIBC assembler to learn the counter-intuitive technique of, to narrow bytes to nibbles, to start by making byte-pairs to shift them by 4 bits (a nibble) so the upper and lower nibbles straddle the byte boundary, then reduce to bytes, which in reality is a pair of nibbles.
If I indicate what comments were mine, sure, but when does it stop? do I also indicate my prompts? what about my user configuration, so that people can replicate exactly? then I'd also have to indicate the day when I had the session, since the tool can change radically one day to the next.